src/modules/manual_control/ManualControl.cpp

```diff
@@ -453,7 +453,7 @@ void ManualControl::send_camera_mode_command(CameraMode camera_mode)
 	command.command = vehicle_command_s::VEHICLE_CMD_SET_CAMERA_MODE;
 	command.param2 = static_cast<float>(camera_mode);
 	command.target_system = _system_id;
-	command.target_component = 100; // any camera
+	command.target_component = 100; // MAV_COMP_ID_CAMERA
```

Comment:
Right, I remember discussing this for various commands in the past.

It turns out that some commands encode it in `param1`, e.g.
https://mavlink.io/en/messages/common.html#MAV_CMD_IMAGE_START_CAPTURE

But SET_MODE does not:
https://mavlink.io/en/messages/common.html#MAV_CMD_SET_CAMERA_MODE

We could add it, but it's a one by one process of adding it.

Reply:
Yes. The update in the PR is to add the id for those commands that support it now - those were added in a few months ago. I think we should add it to camera mode, and if you agree, will add a PR to add for the missing cases next week (?).

Previous discussions have suggested having a generic target ID setter that you change before calling an API. This mostly makes sense because some commands have few free params. I don't like that design as it makes the API more complicated. We'll address it when we come to it.

Reply:
@julianoes OK, so I added support for `MAV_CMD_SET_CAMERA_SOURCE` as well, everywhere except the caching code - for example in `MissionBase::cacheItem(const mission_item_s &mission_item)` - I was wondering if you could tidy that bit because I am concerned I would screw it up. Will do it if you are not happy to take that on.

Reply:
PS Finding out the other commands to add this to in https://github.com/mavlink/mavlink/pull/2111

Reply:
OK, as per discussion in https://github.com/mavlink/mavlink/pull/2111 the only other command to add for this PR was SET_CAMERA_MODE (DONE)